app/src/main/java/de/westnordost/streetcomplete/MainActivity.kt

```diff
@@ -115,6 +116,10 @@ class MainActivity : AppCompatActivity(),
         handleGeoUri()
     }
 
+    override fun attachBaseContext(newBase: Context?) {
+        super.attachBaseContext(LocaleContextWrapper.wrap(newBase))
+    }
+
```

Comment:
I'd really like to avoid something like this. This is a big potential for bugs because one got to remember to add this to every single Fragment and Activity(?)

There are a lot more Fragments and Activities in the app than for which you added those lines. Also, there may be Fragments from third parties (libraries) etc.

Why is this necessary? The code in `LocaleContextWrapper.wrap` contains code that replaces the default locale. This is a static variable, meaning it should stay this way until the app is restarted, right? So, why set it again and again on attaching the base context of each `Activity` and `Fragment`?

I was hoping it would be possible to for example set the default locale within the app to something else **on app startup before any activity is started** (in the `Application` class). And whenever one changes the language setting, the app would restart. Is this not possible?

Reply:
okay, I believe I did not try it in the application class. I could work.
I believe I tried in the main activity and I believe the locale was reset on rotation or something like this.

Fragments from libraries, were this cannot be added, are a good point.